src/libs/actions/Welcome.js

```diff
@@ -116,25 +116,33 @@ function show({routes, showCreateMenu = () => {}, showPopoverMenu = () => {}}) {
         const isWorkspaceRoute = topRoute.name === 'Settings' && topRoute.params.path.includes('workspace');
         const transitionRoute = _.find(routes, (route) => route.name === SCREENS.TRANSITION_BETWEEN_APPS);
         const exitingToWorkspaceRoute = lodashGet(transitionRoute, 'params.exitTo', '') === 'workspace/new';
+        const openOnAdminRoom = lodashGet(topRoute, 'params.openOnAdminRoom', false);
         const isDisplayingWorkspaceRoute = isWorkspaceRoute || exitingToWorkspaceRoute;
 
-        // We want to display the Workspace chat first since that means a user is already in a Workspace and doesn't need to create another one
-        const workspaceChatReport = _.find(
-            allReports,
-            (report) => ReportUtils.isPolicyExpenseChat(report) && report.ownerAccountID === currentUserAccountID && report.statusNum !== CONST.REPORT.STATUS.CLOSED,
-        );
-        if (workspaceChatReport && !isDisplayingWorkspaceRoute) {
-            // This key is only updated when we call ReconnectApp, setting it to false now allows the user to navigate normally instead of always redirecting to the workspace chat
-            Onyx.set(ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER, false);
-            Navigation.navigate(ROUTES.getReportRoute(workspaceChatReport.reportID));
-
-            // If showPopoverMenu exists and returns true then it opened the Popover Menu successfully, and we can update isFirstTimeNewExpensifyUser
-            // so the Welcome logic doesn't run again
-            if (showPopoverMenu()) {
-                isFirstTimeNewExpensifyUser = false;
+        // If we already opened the workspace settings or want the admin room to stay open, do not
+        // navigate away to the workspace chat report
+        const shouldNavigateToWorkspaceChat = !isDisplayingWorkspaceRoute && !openOnAdminRoom;
+
+        if (shouldNavigateToWorkspaceChat) {
+            // We want to display the Workspace chat first since that means a user is already in a Workspace and doesn't need to create another one
+            const workspaceChatReport = _.find(
+                allReports,
+                (report) => ReportUtils.isPolicyExpenseChat(report) && report.ownerAccountID === currentUserAccountID && report.statusNum !== CONST.REPORT.STATUS.CLOSED,
+            );
+
+            if (workspaceChatReport) {
```

Comment:
I think we might want to move this outside of the `shouldNavigateToWorkspaceChat` loop, and update the condition to `if (workspaceChatReport || openOnAdminRoom)`, since we would want to set the NVP and show the popover on the first visit, even if they're going to the admins room.

Does that sound right?

Reply:
As far as I understand your suggestion, changing it to `if (workspaceChatReport || openOnAdminRoom)` will cause it to have the old behaviour of navigating away.

If we want to show the popover but not navigate away, what about this approach? Maybe that's what you meant.

```
        const isDisplayingWorkspaceRoute = isWorkspaceRoute || exitingToWorkspaceRoute;

        const workspaceChatReport = _.find(
            allReports,
            (report) => ReportUtils.isPolicyExpenseChat(report) && report.ownerAccountID === currentUserAccountID && report.statusNum !== CONST.REPORT.STATUS.CLOSED,
        );

        // If we already opened the workspace settings or want the admin room to stay open, do not
        // navigate away to the workspace chat report
        if (workspaceChatReport && !openOnAdminRoom && !isDisplayingWorkspaceRoute) {
             // This key is only updated when we call ReconnectApp, setting it to false now allows the user to navigate normally instead of always redirecting to the workspace chat
            Onyx.set(ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER, false);
            Navigation.navigate(ROUTES.getReportRoute(workspaceChatReport.reportID));
        }
        
        if (workspaceChatReport || openOnAdminRoom) {
           // If showPopoverMenu exists and returns true then it opened the Popover Menu successfully, and we can update isFirstTimeNewExpensifyUser
            // so the Welcome logic doesn't run again
           if (showPopoverMenu()) {
                isFirstTimeNewExpensifyUser = false;
           }
           return;
        }
        
        // ....
```

Reply:
Ah sorry my selection was bad. I meant only moving the `Onyx.set` and the `if (showPopoverMenu())`

I think we could do something like:
```
// All code up to this point is the same as what you have already

if (shouldNavigateToWorkspaceChat) {
	// This is the same loop as you have it currently, but removing the lines below
}

if (workspaceChatReport || openOnAdminRoom) {
	// This key is only updated when we call ReconnectApp, setting it to false now allows the user to navigate normally instead of always redirecting to the workspace chat
	Onyx.set(ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER, false);
	
	// If showPopoverMenu exists and returns true then it opened the Popover Menu successfully, and we can update isFirstTimeNewExpensifyUser
	// so the Welcome logic doesn't run again
	if (showPopoverMenu()) {
	    isFirstTimeNewExpensifyUser = false;
	}
}	
```

So a little simpler than what you suggested but pretty much the same effect.

Reply:
All right good to know, I suggested something else because I thought changing the order of calling `Onyx.set` and `Navigation.navigate` made a difference.

Though, do we really want to show the popover on the admin chat? There is one comment before calling `Welcome.show`: `Shows Popover Menu on Workspace Chat at first sign-in`.

## Issue

<img width="1439" alt="image" src="https://github.com/Expensify/App/assets/5003405/aeeeac42-c72e-49ca-8afb-4396db503cc8">


**If we still want to show the popover on the admin chat, one issue occurs.** We will have to check whether `isWorkspaceRoute` is true, so we don't show the popup while the workspace settings are open (that looks a bit off). However, I don't think that the current value of `isWorkspaceRoute` really makes sense.

https://github.com/Expensify/App/blob/70c525952d33563d88efbaf799b840ab769716ed/src/libs/actions/Welcome.js#L113-L116

## My reasoning

There are two components that call `Welcome.show`. One is `FloatingActionButtonAndPopover` (responsible for the sidebar create menu) and the other one is `ReportActionCompose` (responsible for the chat popover menu).

`FloatingActionButtonAndPopover` only calls `Welcome.show` if the active route is within the central pane navigator or the home screen. So, in this case it is not relevant. When having the workspace settings open, the right modal navigator is active. So, if we are rendering the SidebarScreen at the same time as a workspace route (as the comment above suggests), `Welcome.show` will not be called. See here:

https://github.com/Expensify/App/blob/61ab4bcf100500ce49bf1e2dfaa128f9511ea941/src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js#L153-L156

`ReportActionCompose` does not care about the active navigator and will always execute `Welcome.show` if a report is displayed. However, it is used within the report screen and only has access to the central pane navigator whose routes it will forward to `Welcome.show`. Its only route will be the admin report, even though globally the workspace route is active. So I don't see a scenario in which `isWorkspaceRoute` will be true.

https://github.com/Expensify/App/blob/61ab4bcf100500ce49bf1e2dfaa128f9511ea941/src/pages/home/report/ReportActionCompose.js#L260C6-L266

## Possible resolutions

1. Do not show popover on the admin chat. I don't think it the chat will open without the workspace settings anyway.
2. Use `navigator.getParent()` to access the root navigator within `ReportActionCompose` and do not run `Welcome.show` in case the right modal navigator is active (just like it was done for `FloatingActionButtonAndPopover`).

@dangrous What are your thoughts here?

Reply:
Okay so we haven't heard from QA, I think it's fine not to show the pop up.

But, we SHOULD still set these two lines even when going to the admin room, I think:
`Onyx.set(ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER, false);`
and
`isFirstTimeNewExpensifyUser = false;`

Do you think that makes sense? If so, can you make that update and then we can merge?

Reply:
That makes sense to me. I updated the code so `Onyx.set(ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER, false);` and `isFirstTimeNewExpensifyUser = false;` will be called when going to the admin room and the logic is the same as before otherwise.